### PR TITLE
Syncing with v0.15.0 of Firecracker

### PR DESCRIPTION
**Pull request by xibz**
Two changes were required in updating to v0.15.0 in that the first changed some values to be pointers due to allowing them to have default values. 
There is now validation that the drive name has to be alphanumeric and underscores. Due to this the root drive name, root-drive, has been renamed to root_drive.

Also fixes a bug where `StopVMM` was racing in `TestMicroVMExecution` and would sometimes succeed because the defer would happen last and do nothing with the error. However, if the defer happened before the `TestStopVMM`, then that's when the error is captured and fails.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by samuelkarp** [comment]
Our CI is not currently testing Firecracker v0.15.0.  Do you want help getting the CI to use the new version?

**Comment by xibz** [comment]
Yea, I would love to do that.

File: machine.go [hunk 1]
```diff
@@ -397,11 +397,11 @@ func (m *Machine) setupLogging(ctx context.Context) error {
...
-		LogFifo:       m.cfg.LogFifo,
-		Level:         m.cfg.LogLevel,
-		MetricsFifo:   m.cfg.MetricsFifo,
-		ShowLevel:     true,
-		ShowLogOrigin: false,
+		LogFifo:       String(m.cfg.LogFifo),
+		Level:         String(m.cfg.LogLevel),
+		MetricsFifo:   String(m.cfg.MetricsFifo),
+		ShowLevel:     Bool(true),
+		ShowLogOrigin: Bool(false),
```
Comment (nmeyerhans) [inline machine.go]: The commit message doesn't shed any light on why we need these changes. Are they actually required as part of updating to support v0.15.0?

Comment (xibz) [reply machine.go]: Yea, the newly generated swagger models changed those to be pointers. I think it has something to do with setting default values and required fields, ie if a field isn't set, nil, set to default value.

**Pull request closed without merging**